### PR TITLE
DEP: use of numeric sat_id to specify number of points

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #686
Complements #735.  Moves those warnings here as part of 2.3.0.

I've included notes in the docs throughout the instruments for usage.  Let me know if these should be done differently.

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

testing via travis CI

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +23,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
       - the pysatMadrigal: jro_isr, dmsp_ivm
       - the pysatSpaceWeather: sw_dst, sw_f107, sw_kp
       - the pysatModels: ucar_tiegcm
+   - Test instruments
+     - The usage of a numeric string for `sat_id` to specify number of points
```
Comment (aburrell) [inline CHANGELOG.md:27]: Does this need to be in a sublist when there's only one item?

File: pysat/instruments/pysat_testing.py [hunk 2]
```diff
@@ -119,6 +119,11 @@ def load(fnames, tag=None, sat_id=None, sim_multi_file_right=False,
         If True, the loaded file list time index is shifted by 5-minutes.
         This shift is actually performed by the init function.
 
+    .. deprecated:: 2.3.0
```
Comment (aburrell) [inline pysat/instruments/pysat_testing.py:122]: In other deprecations like this, I've added the new kwarg, so they can use either method for this release.  Recommend doing that here, and in the rest of the testing instruments.

File: pysat/tests/test_instruments.py [hunk 3]
```diff
@@ -391,15 +392,22 @@ def test_download_and_load(self):
...
+        fnames = ['2009-01-01.nofile']
+        with warnings.catch_warnings(record=True) as war:
+            uts, index, date = mm_test.generate_times(fnames, '100', freq='1S')
 
-        # self.directory_format = None
-        # self.file_format = None
-        # self.multi_file_day = False
-        # self.orbit_info = None
+        assert len(war) == 1
+        assert war[0].category == DeprecationWarning
```
Comment (aburrell) [inline pysat/tests/test_instruments.py:413]: Need to also test warning message, to ensure the correct deprecation warning was caught.

**Review by aburrell — CHANGES_REQUESTED** [review]

File: pysat/tests/test_instruments.py (hunk 3 above)
Comment (aburrell) [reply pysat/tests/test_instruments.py:413]: Also, I added a deprecation warning evaluation function in instruments.methods.testing that can be used here.

File: pysat/instruments/methods/testing.py [hunk 4]
```diff
@@ -140,6 +141,13 @@ def generate_times(fnames, sat_id, freq='1S'):
     freq : string
         Frequency of temporal output, compatible with pandas.date_range
         [default : '1S']
+    num : int or NoneType
+        Number of times to generate
+
+    .. deprecated:: 2.3.0
+      The ability to use a numeric string as `sat_id` to specify the number
+      of data points has been removed from pysat in the 3.0.0 release and
+      will be replaced by the `num` keyword as an integer)
```
Comment (aburrell) [inline pysat/instruments/methods/testing.py:150]: Move this up above the parameters block

**Review by aburrell — COMMENTED** [review]
I didn't notice last time, but the deprecation warnings in the doc strings elsewhere have all been placed below the description and above the first section block.

Also, do you think it would be worth changing the use of `sat_id` in the unit tests here?  That had been part of my plan originally, since it would make the log smaller.

**Comment by jklenzing** [comment]
> Also, do you think it would be worth changing the use of `sat_id` in the unit tests here? That had been part of my plan originally, since it would make the log smaller.

This is a good idea.  They should be all changed in the latest push.  🤞

**Review by aburrell — APPROVED** [review]
Looks great!

**Pull request merged**